src/autowiring/tweak.h

```diff
@@ -0,0 +1,87 @@
+// Copyright (C) 2012-2015 Leap Motion, Inc. All rights reserved.
+#pragma once
+#include <atomic>
+#include <type_traits>
+
+namespace autowiring {
+  template<typename T>
+  struct tweak {
+    static_assert(std::is_arithmetic<T>::value, "Cannot define a tweakable value that is not arithmetic");
+
+    tweak(void) = default;
+    tweak(const tweak& rhs) = default;
+    tweak(T value) :
+      value(value),
+      last_observed(value)
+    {}
+
+  private:
+    // Current value, advertised both outside and updated
+    volatile T value = T{};
+
+    // Last observed value
+    volatile T last_observed = T{};
+
+  public:
+    // Accessor methods:
+    T get(void) const volatile { return value; }
+    volatile T& get(void) volatile { return value; }
+
+    /// <returns>
+    /// True if the entry is dirty
+    /// </returns>
+    bool is_dirty(void) const volatile {
+      return value != last_observed;
+    }
+
+    /// <summary>
+    /// Updates the assignment version, clearing the dirty flag
+    /// </summary>
+    /// <returns>
+    /// True if this instance was dirty
+    /// </returns>
+    bool clear_dirty(void) volatile {
+      if(value == last_observed)
+        return false;
+      last_observed = value;
+      return true;
+    }
```

Comment:
What is the intended use of "is_dirty()" and "clear_dirty()"? Since value is both assigned to and returned, it appears to mat that there is no guarantee in a multi-threaded application that the state of a tweak will be clean following a call to "clear_dirty".

What am I missing? And, can you write a unit test to demonstrate the utility of these methods?